lib/src/executables/boilerplate_upgrade.dart

```diff
@@ -44,23 +45,45 @@ const _changesRequiredOutput = '''
   To update your code, run the following commands in your repository:
   pub global activate over_react_codemod
   pub global run over_react_codemod:boilerplate_upgrade
+
+  Or you can use the -p flag to specify a path or glob to run the codemod on only some files:
+  pub global run over_react_codemod:boilerplate_upgrade -p "path/to/your/file.dart"
+  pub global run over_react_codemod:boilerplate_upgrade -p "lib/**.dart"
+  
   pub run dart_dev format (if you format this repository).
   Then, review the the changes, address any FIXMEs, and commit.
 ''';
 
 void main(List<String> args) {
   final parser = ArgParser()
+    ..addFlag('help',
+        abbr: 'h', negatable: false, help: 'Prints this help output')
+    ..addOption('path',
+        abbr: 'p',
+        help:
+            'Specify a path or glob to run codemod on just Dart files specified')
     ..addSeparator('Boilerplate Upgrade Options:')
-    ..addFlag(_convertedClassesWithExternalSuperclassFlag)
-    ..addFlag(_treatAllComponentsAsPrivateFlag)
+    ..addFlag(_convertedClassesWithExternalSuperclassFlag,
+        help: 'Converts classes with external superclasses')
+    ..addFlag(_treatAllComponentsAsPrivateFlag,
+        help: 'Treats all components as private')
     ..addSeparator('Dependency Version Updates:')
-    ..addOption(_overReactVersionRangeOption, defaultsTo: overReactVersionRange)
+    ..addOption(_overReactVersionRangeOption,
+        defaultsTo: overReactVersionRange,
+        help: 'Sets over_react version range')
     ..addOption(_overReactTestVersionRangeOption,
-        defaultsTo: overReactTestVersionRange);
+        defaultsTo: overReactTestVersionRange,
+        help: 'Sets over_react_test version range');
+
   final parsedArgs = parser.parse(args);
 
   final logger = Logger('over_react_codemod.boilerplate_upgrade');
 
+  if (parsedArgs['help'] == true) {
+    stderr.writeln(parser.usage);
+    return;
+  }
+
   exitCode = upgradeReactVersions(
     args: parsedArgs.rest,
```

Comment:
This `args` param gets passed on to `runInteractiveCodemod()`'s `args` param, which is intended to make it possible for consumers to setup a codemod executable without having to do any of its own arg parsing, e.g.:

```dart
main(List<String> args) {
  exitCode = runInteractiveCodemod(..., args: args);
}
```

But in this case, the codemods have their own arg parsing implemented, which means that any flags/options not explicitly added to the above parser are unsupported. For example, a consumer of this codemod cannot pass in the `--yes-to-all` flag that is supported by `runInteractiveCodemod` because the above call to `parser.parse(args)` would throw when it is present. If we wanted to support these options that `runInteractiveCodemod` supports, we'd have to add them to the arg parser used above (or maybe export `codemodArgParser` from `package:codemod` so that it can be extended for use cases like this).

In other words, passing `parsedArgs.rest` here isn't doing anything.

---

All of that is to say, I think it'd be better to use `parsedArgs.rest` as a way to allow consumers to specify exactly which files to run the codemod on instead of the `-p|--path` option, since that's what the basic args are traditionally used for and it allows consumers to utilize their own shell's expansion to do something like this:

```shell
pub global run over_react_codemod:boilerplate_upgrade lib/**.dart
```

This also simplifies the implementation here slightly because you can treat the args as a simple list of file paths.

Reply:
Thanks for this. I pushed a commit to change the pattern to use basic args instead, and removed the doing-nothing `parsedArgs.rest` params.

I'm not too concerned about addressing passing dart-codemod-specific options through correctly to `runInteractiveCodemod` -- it's never come up, and the vast majority of the usage of this codemod is behind us.